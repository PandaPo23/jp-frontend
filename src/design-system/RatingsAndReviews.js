import React from 'react';

import Flex from './Flex';
import RatingControl from './RatingControl';
import ReviewButton from './ReviewButton';

const RatingsAndReviews = ({
  size = 18,
  label,
  initialRating,
  color,
  fullColor,
  ml,
  onClickReview,
  fractions,
  emptyIcon,
  fullIcon,
  readonly,
  py = 0,
}) => (
  <Flex alignItems="baseline" py={py}>
    <RatingControl
      size={size}
      initialRating={initialRating}
      color={color}
      fullColor={fullColor}
      fractions={fractions}
      emptyIcon={emptyIcon}
      fullIcon={fullIcon}
      readonly={readonly}
    />
    <ReviewButton size={size} label={label} ml={ml} onClick={onClickReview} />
  </Flex>
);

export default RatingsAndReviews;
