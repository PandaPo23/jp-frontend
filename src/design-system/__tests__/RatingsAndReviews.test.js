import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RatingsAndReviews from '../RatingsAndReviews';

describe('RatingsAndReviews component', () => {
  it('renders correctly', () => {
    const localProps = {
      label: '6',
      initialRating: 3,
      fullColor: 'secondaryVariant',
      color: 'on.surface',
      ml: 3,
    };
    const wrapper = shallow(<RatingsAndReviews {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
