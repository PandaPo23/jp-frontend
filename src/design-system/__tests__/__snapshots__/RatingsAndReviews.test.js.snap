// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingsAndReviews component renders correctly 1`] = `
<Flex
  alignItems="baseline"
  display="flex"
  py={0}
>
  <RatingControl
    color="on.surface"
    direction="ltr"
    emptyIcon="star-border"
    emptySymbol={
      Object {
        "backgroundColor": "#ccc",
        "border": "5px double white",
        "borderRadius": "50%",
        "display": "inline-block",
        "height": 30,
        "width": 30,
      }
    }
    fractions={1}
    fullColor="secondaryVariant"
    fullIcon="star"
    fullSymbol={
      Object {
        "backgroundColor": "black",
        "border": "5px double white",
        "borderRadius": "50%",
        "display": "inline-block",
        "height": 30,
        "width": 30,
      }
    }
    initialRating={3}
    onChange={[Function]}
    onClick={[Function]}
    onHover={[Function]}
    placeholderSymbol={
      Object {
        "backgroundColor": "red",
        "border": "5px double white",
        "borderRadius": "50%",
        "display": "inline-block",
        "height": 30,
        "width": 30,
      }
    }
    quiet={false}
    readonly={false}
    size={18}
    start={0}
    step={1}
    stop={5}
  />
  <ReviewButton
    color="misc.muted"
    label="6"
    labelMargin={2}
    ml={3}
    name="review-icon"
    size={18}
  />
</Flex>
`;
